core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/policy/PolicyArchiveImpl.java

```diff
@@ -32,8 +32,27 @@ public PolicyArchiveImpl(ContractNegotiationStore contractNegotiationStore) {
     public Policy findPolicyForContract(String contractId) {
         return Optional.ofNullable(contractId)
                 .map(contractNegotiationStore::findContractAgreement)
-                .map(ContractAgreement::getPolicy)
+                .map(this::mapAgreementPolicy)
                 .orElse(null);
     }
 
+    // TODO assignee and assigner should end up stored in the Agreement's policy as outlined here
+    //  https://github.com/International-Data-Spaces-Association/ids-specification/issues/195
+    //  As fallback we fill the assignee and the assigner from the consumer and provider id in
+    //  the contract agreement
```

Comment:
should this deserve a different issue/PR? can be separated?

Reply:
Yes i will raise an issue for supporting this